components/clarinet-format/src/formatter/mod.rs

```diff
@@ -0,0 +1,1220 @@
+                acc.push_str(&self.display_pse(comment, previous_indentation));
+            }
+        }
+        acc.push_str(&format!("\n{})", previous_indentation));
+        acc
+    }
+
+    fn format_list(&self, exprs: &[PreSymbolicExpression], previous_indentation: &str) -> String {
+        let mut acc = "(".to_string();
+        for (i, expr) in exprs[0..].iter().enumerate() {
+            let value = self.format_source_exprs(&[expr.clone()], previous_indentation);
+            if i < exprs.len() - 1 {
+                acc.push_str(&value.to_string());
+                acc.push(' ');
+            } else {
+                acc.push_str(&value.to_string());
+            }
+        }
+        acc.push(')');
+        t(&acc).to_string()
+    }
+
+    // used for { n1: 1 } syntax
+    fn format_key_value_sugar(
+        &self,
+        exprs: &[PreSymbolicExpression],
+        previous_indentation: &str,
+    ) -> String {
+        let indentation = &self.settings.indentation.to_string();
+        let space = format!("{}{}", previous_indentation, indentation);
+        let over_2_kvs = without_comments_len(exprs) > 2;
+        let mut acc = "{".to_string();
+
+        if over_2_kvs {
+            acc.push('\n');
+            let mut counter = 1;
+            for expr in exprs.iter() {
+                if is_comment(expr) {
+                    acc.push_str(&format!(
+                        "{}{}\n",
+                        space,
+                        self.format_source_exprs(&[expr.clone()], previous_indentation)
+                    ))
+                } else {
+                    // if counter is even we're on the value
+                    if counter % 2 == 0 {
+                        // Pass the current indentation level to nested formatting
+                        let value_str = self.format_source_exprs(&[expr.clone()], &space);
+                        acc.push_str(&format!(": {},\n", value_str));
+                    } else {
+                        // if counter is odd we're on the key
+                        acc.push_str(&format!(
+                            "{}{}",
+                            space,
+                            self.format_source_exprs(&[expr.clone()], previous_indentation)
+                        ));
+                    }
+                    counter += 1
+                }
+            }
+            acc.push_str(previous_indentation);
+        } else {
+            // for cases where we keep it on the same line with 1 k/v pair
+            let fkey = self.display_pse(&exprs[0], previous_indentation);
+            acc.push_str(&format!(
+                " {fkey}: {} ",
+                self.format_source_exprs(&[exprs[1].clone()], previous_indentation)
+            ));
+        }
+
+        acc.push('}');
+        acc
+    }
+
+    // used for (tuple (n1  1)) syntax
+    // Note: Converted to a { a: 1 } style map
+    // TODO: This should be rolled into format_key_value_sugar, but the PSE
```

Comment:
that's alright imo. I feel like in certain context (asts manipulation), a bit a duplication is better than to much abstraction